File: client/apollo/js/Projection/AbstractProjection.ts
///<reference path="Coordinate.ts"/>
abstract class AbstractProjection implements ProjectionInterface{

    public static UNMAPPED_VALUE : number = -1;

    abstract projectValue(input:number):number ;
    abstract projectReverseValue(input:number):number ;
    abstract projectSequences(inputSequence:string, min:number, max:number, offset:number):string ;
    abstract getLength():number ;

    projectCoordinate(min:number, max:number):Coordinate {
        return new Coordinate(this.projectValue(min),this.projectValue(max));
    }

    projectReverseCoordinate(min:number, max:number):Coordinate {
        return new Coordinate(this.projectReverseValue(min),this.projectReverseValue(max))
    }

    clear():number {
        return 0;
    }
}

File: src/gwt/org/bbop/apollo/gwt/client/projection/AbstractProjection.java
package org.bbop.apollo.gwt.client.projection;


/**
 * Created by nathandunn on 10/10/16.
 */
public abstract class AbstractProjection implements ProjectionInterface{

    public final static Integer UNMAPPED_VALUE = -1;

    @Override
    public Coordinate projectCoordinate(int min, int max) {
        return new Coordinate(projectValue(min),projectValue(max));
    }

    @Override
    public Coordinate projectReverseCoordinate(int min, int max) {
        return new Coordinate(projectReverseValue(min),projectReverseValue(max));
    }

    @Override
    public Integer clear() {
        return 0;
    }
}
